sqlmesh/core/linter/definition.py

```diff
@@ -108,9 +108,10 @@ def check_model(self, model: Model, context: GenericContext) -> t.List[RuleViola
 
         for rule in self._underlying.values():
             violation = rule(context).check_model(model)
-
+            if isinstance(violation, RuleViolation):
+                violation = [violation]
             if violation:
-                violations.append(violation)
+                violations.extend(violation)
```

Comment:
This looks wrong. Are we adding the same violation twice? Haven't we added it on line 112 already?

Reply:
No it turns the `RuleViolation` into a `t.List[RulationViolation]` before adding it. and if it's None it doesn't get extended.